test/python/test_roles_api.py

```diff
@@ -4,24 +4,714 @@
 
 import openapi_client
 
-from . import api_config
+import api_config
+
+ROLES_POLICY = """
+- !layer testLayer
+
+- !group
+  id: userGroup
+  annotations:
+    editable: true
+
+- !group
+  id: anotherGroup
+  annotations:
+    editable: true
+
+- !user bob
+- !user alice
+"""
+
+NULL_BYTE = '\00'
 
 class TestRolesApi(api_config.ConfiguredTest):
     """RolesApi unit test stubs"""
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.ADMIN_ID = f'{cls.account}:user:admin'
+        cls.ALICE_ID = f'{cls.account}:user:alice'
+        cls.BOB_ID   = f'{cls.account}:user:bob'
+        cls.GROUP_ID = f'{cls.account}:group:userGroup'
+        cls.ROOT_ID  = f'{cls.account}:policy:root'
+
     def setUp(self):
-        self.api = openapi_client.api.roles_api.RolesApi(self.client)
+        self.api = openapi_client.RolesApi(self.client)
+        self.bad_auth_api = openapi_client.RolesApi(self.bad_auth_client)
+
+        # set a new root policy including a new group and users Alice and Bob
+        # these will be used to test member addition/deletion
+        policy_api = openapi_client.PoliciesApi(self.client)
+        policy_api.load_policy(self.account, 'root', ROLES_POLICY)
+
+    def tearDown(self):
+        api_config.load_default_policy(self.account, self.client)
```

Comment:
When loading a policy, if a role is defined in the new policy and already exists, it's maintained instead of replaced. Removing this maintains the member relationships from the previous tests.